adr/2021_04_15_using_iri_instead_of_code_id.md

```diff
@@ -0,0 +1,53 @@
+# Using IRI as API resource identifier in requests instead of code/id
+
+* Status: proposed
+* Date: 2021-04-15
+
+## Context and Problem Statement
+
+API Platform recommends using IRI as an identifier. That identifier gives more clarity than an id because it contains more information - a full endpoint path to the resource and its unique identifier. On resources, API Platform handles IRI out of the box.
+While we were designing Sylius new API, we have decided to use commands in many endpoints.
+In some cases, this solution is more flexible than the default creation approach (as we have control over what we are processing), but it doesn't handle the transformation of IRI to proper `code`/`id`.
+In the past, we have been using `code`/`id` instead of IRI, then we have been using both approaches.
+Now we are trying to unify the new API and replace codes and ids with IRI everywhere.
+The main challenge is the usage of IRI in requests, where we want to have IRI in the request but `id`/`code`in its commands and command handlers.
+
+## Considered Options
+
+### Using `id`/`code` instead of iri
+
+Using `code`/`id` instead of iri is easier implementable, but it is not consistence with most cases.
+ApiPlatform handles it out of the box. Main problem is that for getting fe. `code` we need call "extra" request only for 
+getting code from this object. Iri are available as a collections element or field on certain object. Is many examples
+when we don't need to call next request for getting `code`/`id`.
```

Comment:
```suggestion
Using `code`/`id` instead of IRI is easier to implement, but it is not consistent with API Platform and its default behaviour.
```
I don't understand the second part of this section, what additional request are you talking about?